lib/spraylist_linden/ssalloc.c

```diff
@@ -30,14 +30,20 @@ ssalloc_set(void* mem)
 #endif
 }
 
+extern int number_of_threads;
+
 void
 ssalloc_init()
 {
 #if !defined(SSALLOC_USE_MALLOC)
   int i;
   for (i = 0; i < SSALLOC_NUM_ALLOCATORS; i++)
     {
+#ifdef SSALLOC_SIZE_ALL
+      ssalloc_app_mem[i] = (void*) memalign(64, SSALLOC_SIZE_ALL / number_of_threads);
```

Comment:
I don't think there is much reason not to do that. I will try to remove the global variable.